Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts

```diff
@@ -0,0 +1,201 @@
+            } catch (err) {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('ConnectionFailed', err));
+            }
+
+            if (sshClientConnection) {
+
+                if(playbookSource == 'agentMachine') {
+                    
+                    let playbookRoot: string = tl.getInput('playbookRoot', true);
+                    let playbook: string = tl.getInput('playbookPathAgentMachine', true);
+                    
+                    let remotePlaybookRoot = '/tmp/' + path.basename(playbookRoot);
+                    let remotePlaybookRootPath = '"' + remotePlaybookRoot + '"';
+                    tl.debug('ansiblePlaybookRootPath = ' + remotePlaybookRootPath);
+
+                    let scpConfig = sshConfig;
+                    scpConfig.path = remotePlaybookRoot;
+                    tl.debug('Copying playbook to ansible machine.');
+                    await ansibleUtils.copyFileToRemoteMachine(playbookRoot, scpConfig);
+
+                    let remotePlaybookPath = remotePlaybookRoot + "/" + path.basename(playbook);
+
+                    remoteCommand = remoteCommand.concat(" " + remotePlaybookPath)
+
+                    cleanupCmd.push('rm -rf ' + remotePlaybookRoot);
+
+                } else if(playbookSource == 'ansibleMachine') {
+
+                    let playbook = tl.getInput('playbookPathAnsibleMachine', true);
+                    tl.debug('PlaybookPath = ' + playbook);
+                    remoteCommand = remoteCommand.concat(" " + playbook);
+
+                }
+
+                if(inventoryLocation == 'file') {
+                    
+                    let inventoryFileSource = tl.getInput('inventoryFileSource', true);
+                    if(inventoryFileSource == 'agentMachine') {
+                        let inventoryFile: string = tl.getInput('inventoryFileAgentMachine', true);
+
+                        let remoteInventory = '/tmp/' + path.basename(inventoryFile);
+                        let remoteInventoryPath = '"' + remoteInventory + '"';
+                        tl.debug('RemoteInventoryPath = ' + remoteInventoryPath);
+
+                        let scpConfig = sshConfig;
+                        scpConfig.path = remoteInventory;
+                        tl.debug('Copying Inventory file to ansible machine.');
+                        await ansibleUtils.copyFileToRemoteMachine(inventoryFile, scpConfig);
+
+                        remoteCommand = remoteCommand.concat(' -i ' + remoteInventory);
+
+                        cleanupCmd.push('rm -f ' + remoteInventory);
+
+                    } else if(inventoryFileSource == 'ansibleMachine') {
+
+                        let inventoryFile: string = tl.getInput('inventoryFileAnsibleMachine', true);
+                        tl.debug('InventoryFile = ' + inventoryFile);
+                        remoteCommand = remoteCommand.concat(' -i ' + inventoryFile);

```

Comment:
consider moving this out of both if and else clause and both if and else only set inventoryFile